.github/workflows/main.yml

```diff
@@ -182,11 +182,10 @@ jobs:
         uses: actions/setup-python@v4.3.0
         with:
           python-version: ${{ matrix.python-version }}
-
       - name: Install python dependencies
         run: |
           python -m pip install --user --upgrade pip
-          python -m pip install --upgrade wheel
+          python -m pip install --upgrade wheel setuptools twine check-wheel-contents
```

Comment:
Did adding the following turn out to be necessary?
- setuptools
- twine
- check-wheel-contents

If they aren't strictly necessary, I'd advocate for peeling them out.